jena-arq/Grammar/master.jj

```diff
@@ -327,6 +359,35 @@ void AskQuery() : {}
   SolutionModifier()
 }
 
+void JsonQuery() : {}
+{
+  JsonClause()
+  ( DatasetClause() )*
+  WhereClause()
+  SolutionModifier()
+}
+
+void JsonClause() : { Object o ; String s ; }
+{
+  <JSON> { getQuery().setQueryJsonType() ; }
+  <LBRACE>
+  s = String() < PNAME_NS >
+  (
```

Comment:
I will need some time to check in Eclipse why I used PNAME_NS. If I remember correctly, using COLON, JavaCC would find it ambiguous with some other character or rule?